### ISSUE TITLE
Refactor `explicit_generator`

### ISSUE THREAD

**Issue by noraabiakar** [arbor-sim/arbor#1488]
Right now, it accepts a `pse_vector` so a single generator can connect to multiple targets on the cell.
I propose that we make it more similar to the other generators, which work on a single target at a time.

```
arb::explicit_generator(arb::pse_vector{{0, 1.0, weight_}, {0, 3.0, weight_}, {1, 2.9, weight_}});
```
would become:
```
{arb::explicit_generator(0, weight, {1.0, 3.0}), 
 arb::explicit_generator(1, weight, {2.9})};
```

Meaning that the explicit generator would have the following signature: 

```
event_generator explicit_generator(const cell_lid_type& target, float weight, std::vector<time_type> times);
```

**Comment by halfflat** [comment]
The proposed replacement is essentially equivalent to `schedule_generator` with an explicit schedule.

If we remove the option for an event generator to produce events against different targets, or with different weights, then the type-erased `event_generator` class can just be removed, and replaced with a wrapper for the tuple (_target_, _weight_, _schedule_).

Are there situations where we really want to replay post-synaptic events from a rather arbitrary list? If so, we should probably keep the explicit generator. If not, we can remove `event_generator` and just use something wrapping schedules.

**Comment by noraabiakar** [comment]
Yes, I was thinking along those lines. Except I hadn't looked at all the schedules. We already have what I had in mind implemented as an explicit schedule so that's great. I can do what you suggested and remove the type erased `event_generator`.

As for whether we'd be removing a useful feature, I'm not sure. I think it's quite a specific feature, and it can be replicated by generating a vector of `event_generators`, one per `{target, weight}` pair, with `explicit_schedules`. So we're not removing it, just making it a bit more complex to implement. It's also not a feature that I imagine will be heavily used in large scale simulations where it would make more sense to use the regular or poisson generators, and if it is used, the explicit schedules are not likely to be too big that splitting them across multiple `event_generators` would create inefficiencies in the model initialization.
Can you think of a reason why we would want to keep the current implementation?

**Comment by halfflat** [comment]
I can't think of one off the top of my head, but I'd like to get @bcumming's input before committing to the removal of the general `event_generator`.

**Comment by noraabiakar** [comment]
Okay, let's wait for Ben's input.

**Comment by thorstenhater** [comment]
Hi @bcumming any thoughts?
